### PR TITLE
Feat: follow link prefixed with file:// in the string content using documentLinkProvider

### PR DESCRIPTION
**Pull request by WilsonZiweiWang**
Use the `documentLinkProvider` to provide the correct links in the string content.

### CODE REVIEW

**Comment by WilsonZiweiWang** [comment]
@deribaucourt 
The workflow to provide the document links should be fine in this PR. However, I was encountering some difficulties in finding the files. Had no luck in debugging it.

**Review by deribaucourt — CHANGES_REQUESTED** [review]
I pushed a new commit that fixes vscode.search

File: client/src/documentLinkProvider.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    for (let i = 0; i < linksData.length; i++) {
+      const link = linksData[i]
+      const range = new vscode.Range(
+        link.range.start.line,
+        link.range.start.character,
+        link.range.end.line,
+        link.range.end.character
+      )
+      try {
+        const [file] = await vscode.workspace.findFiles(link.value, undefined, 1)
```
[the remark this answers is not in the document]
Comment (WilsonZiweiWang) [reply client/src/documentLinkProvider.ts]: Your fix is very good.

File: server/src/tree-sitter/analyzer.ts [hunk 2]
```diff
@@ -578,6 +578,49 @@ export default class Analyzer {
...
+              const matchedUri = match.groups?.uri
+              const start = {
+                line: n.startPosition.row + i,
+                character: match.index !== undefined ? match.index + n.startPosition.column : 0
+              }
+              const end = {
+                line: n.startPosition.row + i,
+                character: match.index !== undefined ? match.index + n.startPosition.column + match[0].length : 0
+              }
+              if (i > 0) {
```
[the remark this answers is not in the document]
Comment (WilsonZiweiWang) [reply server/src/tree-sitter/analyzer.ts]: Done.

File: client/src/documentLinkProvider.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (deribaucourt) [reply client/src/documentLinkProvider.ts]: There still was a performance issue in your latest branch. I pushed a commit that greatly reduces the CPU consumption when opening recipes with many defined file:// links.

**Review by deribaucourt — APPROVED** [review]
LGTM, I pushed a commit for performance

**Pull request merged**